frontend/gateway/gateway.go

```diff
@@ -75,7 +75,7 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 
 	_, isDevel := opts[keyDevel]
 	var img specs.Image
-	var rootFS cache.ImmutableRef
+	var rootFS cache.MutableRef
```

Comment:
The previous `llbBridge.Exec` used to do the translation from ImutableRef to MutableRef, but I think that was to suit the usage in gateway for running frontends, so moving that logic here.  I think `llbBridge.Run` generically should allow for immutable rootFS if desired.